src/wire_protocol/header.rs

```diff
@@ -115,7 +115,7 @@ impl Header {
     /// # Return value
     ///
     /// Returns nothing on success, or an Error on failure.
-    pub fn write(&self, buffer: &mut Write) -> Result<()> {
+    pub fn write(&self, buffer: &mut dyn Write) -> Result<()> {
```

Comment:
We should probably change these to be generic rather than use trait objects, but that's not necessary for this PR.

Reply:
Good point, updated